app/src/main/java/wordtextcounter/details/main/util/Helper.kt

```diff
@@ -35,25 +20,107 @@ object Helper {
       while (graphemeCounter.next() != BreakIterator.DONE) {
         graphemeCount++
       }
-
       graphemeCount
     }.subscribeOn(Schedulers.computation())
   }
 
+  fun countCharactersAndSpaces(input: String): Single<Pair<Int, Int>> {
+    return Single.fromCallable {
+      var graphemeCount = 0
+      var whiteSpaceCount = 0
+      val graphemeCounter = BreakIterator.getCharacterInstance()
+      graphemeCounter.setText(input)
+
+      var start = 0
+      var end = graphemeCounter.next()
+      while (end != BreakIterator.DONE) {
+        graphemeCount++
+        if (Character.isWhitespace(input[start])) {
+          whiteSpaceCount++
+        }
+        start = end
+        end = graphemeCounter.next()
+      }
+      Pair(graphemeCount, graphemeCount - whiteSpaceCount)
+    }.subscribeOn(Schedulers.computation())
+  }
+
   fun countSentences(input: String): Single<Int> {
     return Single.fromCallable {
       var graphemeCount = 0
       val boundary = BreakIterator.getSentenceInstance()
       boundary.setText(input)
+      var start = 0
       var end = boundary.next()
       while (end != BreakIterator.DONE) {
-        graphemeCount++
+        val sentence = input.substring(start, end)
+        if (sentence.isNotBlank()) {
+          graphemeCount++
+        }
+        start = end
         end = boundary.next()
       }
       graphemeCount
     }.subscribeOn(Schedulers.computation())
   }
 
+  /**
+   * This will return triple where first is total word count, second is avg word length and third is unique words in text
+   */
+  fun extraWordStats(input: String): Single<Triple<Int, Double, Int>> {
+    return Single.fromCallable {
+      val wordIterator = BreakIterator.getWordInstance()
+      val wordSet = mutableSetOf<String>()
+      wordIterator.setText(input)
+      var start = wordIterator.first()
+      var end = wordIterator.next()
+      var wordCount = 0
+      var totalLength = 0
+      while (end != BreakIterator.DONE) {
+        val word = input.substring(start, end)
+        if (Character.isLetterOrDigit(word[0])) {
+          wordSet.add(word)
+          wordCount++
+          totalLength += word.length
+        }
+        start = end
+        end = wordIterator.next()
+      }
+      Triple(wordCount, totalLength.toDouble() / wordCount, wordSet.size)
+    }.subscribeOn(Schedulers.computation())
+  }
+
+
+  fun extractSentenceStat(input: String): Single<Triple<Int, Int, Int>> {
+    return Single.fromCallable {
+      var minSentenceLength = Int.MAX_VALUE
+      var maxSentenceLength = 0
+      var totalSentences = 0
+      val boundary = BreakIterator.getSentenceInstance()
```

Comment:
Should we move all these Iterator instances at top level in Helper?

Reply:
Not sure. It depends on Locale and as Helper is `object` type, the iterator may be inconsistent if locale is changed during app session